src/alignment/Graph.cpp

```diff
@@ -1,124 +1,146 @@
 	}
-	calcEdgeList();
+    calcEdgeList();
 }
 
-// Method: get adjacent edges from node
-vector<int> Graph::getAdjacentEdge(int seq, int kNum){
-	vector<string>& stringList = getStringListSequence();			
-	vector<int> adjacentNode;
+// Method: get vector of adjacent edges from all nodes in one sequence
+vector<vector<Node>> Graph::getAdjacentEdges(unsigned int index){			
+    vector<vector<Node>> adjacentNodes;
+    vector<Node> emptyVecNode;
+    vector<string>& stringList = getStringListSequence();
 
-	if (seq == stringList.size()-1) {								// the last sequence has no next sequence
-		cerr << kNum << "It is the last sequence." << endl;
-		return adjacentNode;
-	}
-	
-	int kmer = getK(); 
-	int position = kmer * kNum;										
-
-	if (position + kmer <= stringList.at(seq).size()) {				
-		for (int i=0; i < stringList.at(seq+1).size(); i = i+kmer) {// iteration through the sequence
-			if (i+kmer > stringList.at(seq+1).size())
-				break;                         
-			if (stringList.at(seq).substr(position,kmer) == stringList.at(seq+1).substr(i,kmer))
-				adjacentNode.push_back(i/kmer);		
-		}
-		return adjacentNode;
-	} else {
-		cerr << kNum << ".k-mer don't exists in sequence " << seq  << "." << endl;
-		return adjacentNode; 
-	}	
-	cerr << kNum << ".k-mer don't exists in sequence " << seq  << "." << endl;
-	return adjacentNode;
+    // the last sequence has no next sequence
+    if (index == stringList.size()-1) {                                 
+        cerr << "It is the last sequence." << endl;
+        return adjacentNodes;
+    }
+    // calculate adjacent edges from all nodes in one sequence 
+    for (unsigned int i = 0; i < nodeList.at(index).size(); i++) {
+        // initialize adjacentNode
+        adjacentNodes.push_back(emptyVecNode);
+        for (unsigned int j = 0; j < nodeList.at(index + 1).size(); j++) {
+            // compare strings of nodes
+            if (nodeList.at(index).at(i).kmer == nodeList.at(index + 1).at(j).kmer){
+                adjacentNodes.at(i).push_back(nodeList.at(index + 1).at(j));  
+            }            
+        }
+    }
+    return adjacentNodes;
 }
-// Method: get Number of k-mer
-const vector<int>& Graph::getAmountOfKmer(){
-	vector<string>& stringList = getStringListSequence();
-	int sizeOfStringList = stringList.size();
 
-	if (!numberOfKmer.empty()) {
-		return numberOfKmer; 		
-	} else {
-		for (int i = 0; i < sizeOfStringList; i++) {
-			numberOfKmer.push_back(stringList.at(i).size() / getK());						
-		}
-	}
-	return numberOfKmer; 
+// Method: get amount of nodes for all sequences
+const vector<unsigned int>& Graph::getAmountOfKmer(){
+    vector<string>& stringList = getStringListSequence();
+    int sizeOfStringList = stringList.size();
+
+    if (!numberOfKmers.empty()) {
+        return numberOfKmers; 		
+    } else {
+        for (int i = 0; i < sizeOfStringList; i++) {
+            numberOfKmers.push_back(stringList.at(i).size() / getK());						

```

Comment:
Why do you divide by getK()?
This appears to be an uncommented but important assumption that
1) the k-mers are tiled (shift=k) and
2) that all k-mers of the sequence are used in the graph. However, the k-mers that do not have a match in another sequence are isolated nodes without an edge and irrelevant to the task and the game.